apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -2039,431 +2096,49 @@ mod tests {
     use tracing_subscriber::Layer;
 
     use super::apollo::ForwardHeaders;
+    use super::Telemetry;
     use crate::error::FetchError;
     use crate::graphql::Error;
     use crate::graphql::Request;
     use crate::http_ext;
     use crate::json_ext::Object;
+    use crate::metrics::FutureMetricsExt;
     use crate::plugin::test::MockSubgraphService;
     use crate::plugin::test::MockSupergraphService;
     use crate::plugin::DynPlugin;
-    use crate::plugins::telemetry::handle_error;
+    use crate::plugins::telemetry::handle_error_internal;
     use crate::services::SubgraphRequest;
     use crate::services::SubgraphResponse;
     use crate::services::SupergraphRequest;
     use crate::services::SupergraphResponse;
 
```

Comment:
The tests in this file have been improved/split up.
They didn't work with the upgrade due to the way that they relied on the meter provider and metrics layer to be part of the otel plugin. The otel plugin no longer has a reference to the aggregate meter provider.